File: js/src/test/Exchange/base/test.lastPrice.js
import testSharedMethods from './test.sharedMethods.js';
function testLastPrice(exchange, skippedProperties, method, entry, symbol) {
    const format = {
        'info': {},
        'symbol': 'ETH/BTC',
        'timestamp': 1502962946216,
        'datetime': '2017-09-01T00:00:00',
        'price': exchange.parseNumber('1.234'),
        'side': 'buy', // buy or sell
    };
    const emptyAllowedFor = ['timestamp', 'datetime', 'side', 'price']; // binance sometimes provides empty prices for old pairs
    testSharedMethods.assertStructure(exchange, skippedProperties, method, entry, format, emptyAllowedFor);
    testSharedMethods.assertTimestampAndDatetime(exchange, skippedProperties, method, entry);
    //
    testSharedMethods.assertGreater(exchange, skippedProperties, method, entry, 'price', '0');
    testSharedMethods.assertInArray(exchange, skippedProperties, method, entry, 'side', ['buy', 'sell', undefined]);
}
export default testLastPrice;

File: julia/Ccxt/test/validators/test.lastPrice.jl
using Test
using Ccxt
function testLastPrice(exchange, skippedProperties, method, entry, symbol)

    format = Dict{Symbol, Any}(
        Symbol("info") => Dict{Symbol, Any}(),
        Symbol("symbol") => "ETH/BTC",
        Symbol("timestamp") => 1502962946216,
        Symbol("datetime") => "2017-09-01T00:00:00",
        Symbol("price") => parseNumber(exchange, "1.234"),
        Symbol("side") => "buy"
    );
    emptyAllowedFor = ["timestamp", "datetime", "side", "price"];
    assertStructure(testSharedMethods, exchange, skippedProperties, method, entry, format, emptyAllowedFor);
    assertTimestampAndDatetime(testSharedMethods, exchange, skippedProperties, method, entry);
    assertGreater(testSharedMethods, exchange, skippedProperties, method, entry, "price", "0");
    assertInArray(testSharedMethods, exchange, skippedProperties, method, entry, "side", ["buy", "sell", nothing]);
end
